File: examples/R/request_image.R
#!/usr/bin/env Rscript
suppressMessages(library(scrapystreaming))
suppressMessages(library(base64enc))

save_file <- function(response) {
    f <- file("outputs/image.png", "wb")
    base64decode(response$body, f)
    close(f)
    close_spider()
}
create_spider("image", character(0))
send_request("http://httpbin.org/image/png", save_file, TRUE)

run_spider()

File: examples/Nodejs/request_image.js
#!/usr/bin/env node

var scrapy = require('scrapystreaming');
var fs = require('fs');


scrapy.createSpider('image', [], function(response) {});
scrapy.sendRequest('http://httpbin.org/image/png', function(response) {
    fs.writeFile('outputs/image.png', response.body, {encoding: 'base64'});
    scrapy.closeSpider();
},  {base64: true});

scrapy.runSpider();
